server/controllers/analysisController.ts

```diff
@@ -35,21 +35,16 @@ const createAnalysis = async (request: Request, response: Response) => {
 
   console.info('Create analysis for sampleId', sample.id, analysisToCreate);
 
-  const analysis: CreatedAnalysis = {
-    id: uuidv4(),
-    createdAt: new Date(),
+  const analysis: Omit<CreatedAnalysis, 'id'> = CreatedAnalysis.omit({id: true}).parse({
     createdBy: user.id,
     status: 'Residues',
+    createdAt: new Date(),
     ...analysisToCreate,
-  };
-  await analysisRepository.insert(analysis);
-
-  await sampleRepository.update({
```

Comment:
Perso je suis pas fan de déporter de la logique métier dans les repository qui sont pour moi juste une interface avec la base de données et qui devraient se limiter dans la mesure du possible à des méthodes CRUD

Reply:
Tu proposes quoi ?
C'est justement pour palier à un « truc pas cool ». La présence d'une analyse modifie le statut du Sample.
Donc est-ce normal que ça soit en bdd ? Est-ce que c'est pas juste du métier ? Et donc dans le code on pourrait modifier le statut à la volée ?
J'ai pas de solution magique :-/

Reply:
Oui effectivement c'est une erreur de devoir modifier le statut du prélèvement en base de données en fonction du statut de l'analyse. 
Chacun des deux status devrait avoir son propre cycle de vie et à la récupération des prélèvements on devrait avoir un status et un analysisStatus qui permettent de faire l'affichage et le filtrage attendu. 

Je te propose par homogénéité de laisser pour le moment cette logique dans le controller (car on a d'autres endroits ou c'est fait) et de créer une carte tech pour gérer ça proprement de façon globale.

Reply:
Ok ça me va, je vais dupliquer l'update du statut